### DISCUSSION TITLE
Why read byte by byte when it is possible to read all the bytes in can\interfaces\slcan.py?

### DISCUSSION THREAD

**Discussion by silverv** [hardbyte/python-can#1571] [Q&A]
@felixdivo @mikisama 
https://github.com/hardbyte/python-can/blob/af52622b0c62e64f0de99a129a15a74ca84bb4a3/can/interfaces/slcan.py#L164
Is there a specific reason for size=1? Before size=1, in git blame, it was not set and the default was still 1.
As an alternative, that also works, it is possible to read the bytes that are waiting to be read in one go, and hopefully delay less:
`new_bytes = self.serialPortOrig.read(size=self.serialPortOrig.in_waiting)`
I am in the process of measuring the changed delays.

**Comment by felixdivo** [comment]
There was a long discussion in #1490.

🤷 I'm not into that, so I'll retreat here.

**Comment by silverv** [comment]
In continuation to https://github.com/hardbyte/python-can/pull/1490 I created https://github.com/hardbyte/python-can/pull/1572
